docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -82,11 +75,11 @@ Only the following plugins still get updated on every [`updateSettings()`](@/api
 
 #### Migrating to Handsontable 12.0
 
-If you want a plugin (e.g. [`ContextMenu`](@/api/contextmenu.md)) to still get updated every time you call [`updateSettings()`](@/api/core.md#updatesettings), use the [`afterUpdateSettings`](@/api/hooks.md#afterupdatesettings) hook to call your plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method:
+If you want a plugin (e.g. a [custom plugin](@/guides/building-and-testing/plugins.md)) to still get updated every time you call [`updateSettings()`](@/api/core.md#updatesettings), use the [`afterUpdateSettings`](@/api/hooks.md#afterupdatesettings) hook to call your plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method:
 
 ```js
 afterUpdateSettings() {
-  contextMenu.updatePlugin();
+  customPluginInstance.updatePlugin();
```

Comment:
Closer, but I don't think this is the correct advice. We should explain here how to make your custom plugin ready for the new behavior of `updateSettings`, not a hacky way to use the `afterUpdateSettings` hook to trigger the update from the outside

The correct advice is explained in the **Before** and **After** of https://github.com/handsontable/handsontable/pull/9025 OP.

@kirszenbaum if in doubt, please re-consult @jansiegel

Reply:
Maybe let's skip the part about calling the `updatePlugin` method entirely and leave the information about a possibility of utilizing the `afterUpdateSettings` hook to a separate callback then? WDYT @warpech @kirszenbaum ?

Reply:
I don't think that `afterUpdateSettings` is the key information here. The key information is, what do I have to do with my custom plugins like these:

- https://github.com/handsontable/spreadsheet-viewer-dev/blob/develop/src/sv/components/SpreadsheetContents/FloatingBoxPlugin/index.ts
- https://github.com/handsontable/spreadsheet-viewer-dev/blob/develop/src/sv/components/SpreadsheetContents/HorizontalOverlapPlugin/index.ts

How can I make sure that they properly react to the configuration option changes that they depend on?

Reply:
Ok, so focus more on defining the `SETTING_KEYS` static:

@kirszenbaum It's described in this PR -> https://github.com/handsontable/handsontable/pull/9025#issue-1074371739

Basically, the plugin will react to the `updateSettings` call if the config object passed to `updateSettings` contains an entry from their `SETTING_KEYS` static property (which is supposed to be an array or boolean). 

By default, all plugins inheriting from the `BasePlugin` will have the property declared as an array with a single entry - their `PLUGIN_KEY`. If the plugin needs to react to more config options, they need to be added to the array as well. 

Setting the `SETTING_KEYS` as `true` will make the plugin react to _all_ `updateSetting` calls and setting it to `false` will make it _never_ react to them.

Reply:
Thank you, I think this is perfect. This information should be added to the migration guide as well as the custom plugin guide.

Reply:
@jansiegel @warpech So I guess that after all we do need to mention `PLUGIN_KEY` and `SETTING_KEYS` in the migration guide, cause you can't explain the above without them.

Here's my take:
[5d62306](https://github.com/handsontable/handsontable/pull/9271/commits/5d62306a24c7e0a1b1e03ce90c631cbf64cca5f3)
[9253a19](https://github.com/handsontable/handsontable/pull/9271/commits/9253a19be77da6d7b78bd9f5772495a5560c0fb6)

Reply:
Very well! I have provided minor comments as separate threads, resolving here